tools/walletextension/services/sk_manager.go

```diff
@@ -0,0 +1,102 @@
+package services
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/ecies"
+	gethlog "github.com/ethereum/go-ethereum/log"
+	"github.com/ten-protocol/go-ten/go/common/viewingkey"
+	"github.com/ten-protocol/go-ten/tools/walletextension/common"
+	"github.com/ten-protocol/go-ten/tools/walletextension/storage"
+)
+
+type SKManager interface {
+	CreateSessionKey(user *common.GWUser) (*common.GWSessionKey, error)
+	SignTx(ctx context.Context, user *common.GWUser, input hexutil.Bytes) (hexutil.Bytes, error)
+}
+
+type skManager struct {
+	storage storage.UserStorage
+	config  *common.Config
+	logger  gethlog.Logger
+}
+
+func NewSKManager(storage storage.UserStorage, config *common.Config, logger gethlog.Logger) SKManager {
```

Comment:
it would be very helpful to have a quick description of how sessionKeys work. 
The code is very clean, but for someone just trying to check what's going on it will help.